### DISCUSSION TITLE
Opt-In Map Sharing

### DISCUSSION THREAD

**Discussion by jordanbarnes94** [Mydayyy/Valheim-ServerSideMap#9] [Ideas]
For my server I want map sharing to be opt-in: let players choose to share their map data with the server-side map.

I think something similar is discussed in other comments, but I am not looking for separate toggles for map sharing and marker sharing options. Rather, that map storage only happens when players want to share, for example while visibility to other players is enabled.

**Comment by Mydayyy** [comment]
Greetings,

I am not so sure whether I understand this use-case correctly:

The current situation is, that the opt in to sharing is basically installing this plugin. Users who wish not to share their map data would leave this plugin uninstalled. Obviously, with this system you cant really opt-in on a per server basis, except enabling and disabling the plugin all the time which is less than optimal.

In the next update, after the added marker share, I plan to enable opting out of map or marker share separately. But this again would not be on a per server basis since you'd need to change your config file each time you want to enable/disable it.

Your use case seems to be to add a possibility to enable/disable map sharing on a per server basis e.g by using the visibility to other player button. Am I correct on that?


That is certainly an interesting idea to consider, and as of right now I cant think of any use case that would be interfered by that

**Reply by jordanbarnes94** [reply]
Hello,

My apologies, I did not realise the plugin won't take a client's map data if they do not have the plugin. This essentially answers my suggestion to allow clients to opt in or out of map sharing.

I'm glad to hear you plan to add map marker sharing and have it a separate option to map sharing. Thanks for the mod, hope everybody sees its value!
